app/section/sections/mobile-operating-systems/ui.test.js

```diff
@@ -0,0 +1,192 @@
+    .expect(operatingSystemsDescriptionQuestion.find('textarea').count).eql(1)
+    .expect(operatingSystemsDescriptionQuestion.find('[data-test-id="textarea-field-footer"]').innerText).eql(expectedDescription.footerAdvice);
+});
+
+test('should populate the operating systems description textarea with existing data', async (t) => {
+  await pageSetup(t, 200, mobileOperatingSystemsMarketingData);
+  const descriptionQuestion = Selector('[data-test-id="question-operating-systems-description"]');
+
+  await t
+    .expect(descriptionQuestion.find('textarea').value).eql(mobileOperatingSystemsMarketingData['operating-systems-description']);
+});
+
+test('should show error summary and validation for operating systems description question when it exceeds the maxLength', async (t) => {
+  const question = 'operating-systems-description';
+  await pageSetup(t);
+  nock('http://localhost:8080')
+    .put(sectionApiUrl)
+    .reply(400, {
+      maxLength: [question],
+    });
+
+  const errorSummary = Selector('[data-test-id="error-summary"]');
+  const errorSummaryList = Selector('.nhsuk-error-summary__list');
+  const operatingSystemsDescription = Selector(`[data-test-id="question-${question}"]`);
+
+  const submitButton = Selector('[data-test-id="section-submit-button"]');
+  const expectedDescription = sectionManifest.questions[question];
+  await t
+    .expect(errorSummary.exists).notOk()
+    .typeText(operatingSystemsDescription, `${'x'.repeat(501)}`, { paste: true })
+    .click(submitButton.find('button'))
+    .expect(errorSummary.exists).ok()
+    .expect(errorSummaryList.find('li').count).eql(1)
+    .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql(expectedDescription.errorResponse.maxLength)
+    .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql('#operating-systems-description')
+    .expect(operatingSystemsDescription.find('.nhsuk-textarea--error').exists).ok()
+    .expect(operatingSystemsDescription.find('.nhsuk-error-message').innerText).eql(`Error:\n${expectedDescription.errorResponse.maxLength}`);
+});
+
+test('should go to anchor when clicking the operating systems description error link', async (t) => {
+  const question = 'operating-systems-description';
+  await pageSetup(t);
+  nock('http://localhost:8080')
+    .put(sectionApiUrl)
+    .reply(400, {
+      maxLength: [question],
+    });
+  const errorSummaryList = Selector('.nhsuk-error-summary__list');
+  const submitButton = Selector('[data-test-id="section-submit-button"]');
+  await t
+    .click(submitButton.find('button'))
+    .click(errorSummaryList.find('li:nth-child(1) a'))
+    .expect(getLocation()).contains(`/S100000-001/section/${sectionId}#${question}`);
+});
+
+test('should render the submit button', async (t) => {
+  await pageSetup(t);
+  const submitButton = Selector('[data-test-id="section-submit-button"]');
+  await t
+    .expect(submitButton.find('button').count).eql(1);
```

Comment:
Do we not check the content in the other tests?

Reply:
no - I have added here